packages/styled/types/base.ts

```diff
@@ -90,12 +100,24 @@ export interface CreateStyledComponent<
 export interface CreateStyled<Theme extends {} = any> {
   <C extends React.ComponentType<React.ComponentProps<C>>>(
     component: C,
-    options?: StyledOptions
+    options?: StyledOptions<PropsOf<C>>
   ): CreateStyledComponent<PropsOf<C> & { theme?: Theme }, {}, { theme: Theme }>
 
+  <
```

Comment:
I added a new overload to prevent a breaking change of forcing the shouldForwardProp to be a type guard